kiwi/builder/container.py

```diff
@@ -57,6 +60,26 @@ def __init__(self, xml_state, target_dir, root_dir):
         self.target_dir = target_dir
         self.container_config = xml_state.get_container_config()
         self.requested_container_type = xml_state.get_build_type_name()
+        self.base_image_uri = xml_state.get_derived_from_image_uri()
+        self.base_image = None
+        self.base_image_md5 = None
+
+        if self.base_image_uri:
+            base_image_origin = self.base_image_uri.translate()
+
+            # we expect the base image to be unpacked by the kiwi prepare step
+            # and stored inside of the root_dir/image directory. In addition
+            # a md5 file of the image is expected there too
+            self.base_image = os.path.normpath(
+                os.sep.join(
+                    [
+                        root_dir, 'image',
+                        base_image_origin.replace('.tar.xz', '')
+                    ]
+                )
+            )
+            self.base_image_md5 = ''.join([self.base_image, '.md5'])
```

Comment:
Hmmm... this is so conceptual spaghetti! I would ask myself: if I already have a future image name in some sort of config and I end up with `base_image_origin` that contains some extension like `.tar.xz`, then why I am unable to do something like:
```python
# Somewhere earlier in your Universe
if not config.get('image_name'):
    raise Exception('Hey, please tell me where to generate the whole thing!')
# Construct the base_image_origin, based on configured archive
# I am assuming everything goes to the TAR first.
base_image_origin = '{0}.tar.{1}'.format(config.get('image_name'), config.get('archiver_of_the_day'))
# OK so far?

...

# Then here
self.base_image = os.path.join(root_dir, 'image', config.get('image_name'))
self.base_image_chksm = '{0}.{1}'.format(self.base_image, config.get('hashing_algorithm'))
```
I would not assume it is _always_ MD5. What if I want to use SHA256?.. I would not assume it is always one type of compression. What if I want to use BZip2? And so on.